### PR TITLE
Add a list event handler with vi keybindings and a demo.

### PR DESCRIPTION
**Pull request by rhofour**
This adds the vi-keybindings I mentioned in #139 , except the ones for moving around in the visible selection.

ListDemoVi.hs is identical to ListDemo.hs except for line 55 where the vi bindings are added.

### CODE REVIEW

File: programs/ListDemoVi.hs [hunk 1]
```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Main where
+
+import           Control.Monad        (void)
```
Comment (jtdaugherty) [inline programs/ListDemoVi.hs]: Please reformat these import statements without the aligning whitespace.

File: src/Brick/Widgets/List.hs [hunk 2]
```diff
@@ -98,6 +99,30 @@ handleListEvent e theList =
         EvKey KPageUp [] -> listMovePageUp theList
         _ -> return theList
 
+-- | Enable list movement with the vi keys with a fallback if none match.
+-- Movements handled: Up, Down, Page Up, Page Down, Half Page Up, Half Page
```
Comment (jtdaugherty) [inline src/Brick/Widgets/List.hs]: Please document which Vi keys correspond with which behaviors in this Haddock.

Comment (rhofour) [reply src/Brick/Widgets/List.hs]: I was on the fence about this since these are all standard bindings, but I can document them.

Comment (jtdaugherty) [reply src/Brick/Widgets/List.hs]: Thanks! Even if they are standard bindings, documenting them makes this handler consistent with the other list documentation and makes this library easier to use for someone who is *not* already a `vi` user.

**Comment by jtdaugherty** [comment]
Thanks!

**Pull request merged**